### PR TITLE
"Rebase" to geth 1.7.3 and add status geth patches

### LINKED ISSUE #467: Rebase to Geth 1.7.3
Geth 1.7.3 released and we need to sync our[ go-ethereum fork](https://github.com/status-im/go-ethereum) with this version.

Wiki page: https://www.notion.so/status/Rebasing-go-ethereum-82de0bbebe684826b62d81bc688e5e89

https://github.com/ethereum/go-ethereum/releases/tag/v1.7.3

### PR DESCRIPTION
**Pull request by divan**
This PR codifies our changes to the go-ethereum upstream in the form of diffs (.patch file format) and serves as a prerequisite to change our approach to syncronization with geth codebase.
 
 - Patches need to be updated per each geth release
 - They are up to date, so anyone can inspect them and clearly see what exactly we add into geth codebase

It also updates geth dependency in vendor/ to match latest v.1.7.3 release.

- [x] extract previous geth commits from our forked version and codify them as patches
- [x] update patches to work with 1.7.3 geth release
- [x] apply patches and update vendor/ folder
- [x] write scripts (or just doc?) for automatically applying patches for future releases
- [x] cht.json now contains only bootnodes addresses. New CHT/BloomTrie checkpoints are still hardcoded into `go-ethereum` and updated via patches.

README for patches: https://github.com/status-im/status-go/blob/adf7895549037bea5cc287636acb2e1236c56b44/geth-patches/README.md

Closes #467

### CODE REVIEW

File: geth-patches/0000-accounts-hd-keys.patch
Comment (adambabik) [inline geth-patches/0000-accounts-hd-keys.patch:1]: Should we store geth patches here or in our fork of go-ethereum?

File: geth/node/node.go
Comment (adambabik) [inline geth/node/node.go:176]: This is pretty important for us as otherwise LES must be synced from the beginning. How is it handled now? LES/2 feature?

Comment (adambabik) [reply geth/node/node.go:176]: I have just read it in the PR description... 

One way to test it is to run a full node with LES enabled and try to sync your local node using that full node as a peer. We have a full node in our cluster. Let me know if you don't have access.

File: geth-patches/0000-accounts-hd-keys.patch
Comment (divan) [reply geth-patches/0000-accounts-hd-keys.patch:1]: @adambabik where to store patches is an open question, but with this approach we don't need to keep "our fork" repo anymore. Patches will be applied from scratch on each release (I believe this process could be automated via simple scripting)

**Comment by divan** [comment]
Related
https://github.com/ethereum/go-ethereum/issues/15575
https://github.com/ethereum/go-ethereum/issues/15482

**Comment by divan** [comment]
After upgrading one of the Status bootnodes to Geth 1.7.3, and testing this PR, I can see that synchronization on Ropsten is starting from trusted checkpoint, hardcoded in Geth 1.7.3 code (and light mode sync takes around 5 mins on my laptop). This release seems to have implemented CHT functionality and my assumption is that trusted checkpoint should be updated regularly. Holding this PR until this moment is clarified.

**Review by adambabik — COMMENTED** [review]
Where are CHT checkpoints hardcoded now?

**Review by themue — APPROVED** [review]
It's a limited LGTM as I'm not yet deep enough into the individual patches. But I like the approach and the realisation.

A final LGTM should come by @adambabik or @tiabc .

**Comment by divan** [comment]
> Where are CHT checkpoints hardcoded now?

@adambabik they're implemented as a patch, which will be updated and re-applied from time to time until we implement proper Trusted Checkpoints mechanism: https://github.com/status-im/status-go/pull/492/files#diff-baeffc2b5580436f546bac21a047f513

Also, make sure you've seen this README for `geth-patches` dir, it also a good subject for review: https://github.com/status-im/status-go/blob/adf7895549037bea5cc287636acb2e1236c56b44/geth-patches/README.md

File: geth-patches/0006-latest-cht.patch
Comment (adambabik) [inline geth-patches/0006-latest-cht.patch:10]: I assume it has been calculated recently, right? When testing locally how long does it take to sync to the most recent block?

File: geth-patches/README.md
Comment (adambabik) [inline geth-patches/README.md:39]: We could also recommend adding `git remote set-url upstream git@github.com:ethereum/go-ethereum.git` so any further pulls from upstream could be done like `git pull upstream release/1.7:master`.

**Review by adambabik — APPROVED** [review]

File: geth-patches/0006-latest-cht.patch
Comment (divan) [reply geth-patches/0006-latest-cht.patch:10]: Yes, I did it a couple of days ago and sync took 9-10 seconds on my laptop.

File: geth-patches/README.md
Comment (divan) [reply geth-patches/README.md:39]: Great! Feel free to update README. I like the way you proposed — it's more explicit, but can't check it now from scratch, so let's update this after this PR is merged. Both options work, right?

Comment (adambabik) [reply geth-patches/README.md:39]: It sounds good. I will make a PR when this one is merged. Also, I will test the whole flow shortly after this PR is merged as I have a use-case for a new patch.

**Review by tiabc — CHANGES_REQUESTED** [review]
Please, let me review the changes before merging.

**Comment by divan** [comment]
This PR is needed to be merged asap as it's blocking other PRs. I'm setting a deadline for tomorrow (Jan 9th). @tiabc if you have a chance to review it today, it would be great.

**Review by tiabc — APPROVED** [review]

**Pull request merged**
